diff --git forkSrcPrefix/audio/out/ao_coreaudio_utils.h forkDstPrefix/audio/out/ao_coreaudio_utils.h
--- forkSrcPrefix/audio/out/ao_coreaudio_utils.h
+++ forkDstPrefix/audio/out/ao_coreaudio_utils.h
@@ -53,7 +53,7 @@
     } while (0)
 
 void ca_get_device_list(struct ao *ao, struct ao_device_list *list);
-#if HAVE_COREAUDIO || HAVE_AVFOUNDATION
+#if HAVE_COREAUDIO
 OSStatus ca_select_device(struct ao *ao, char* name, AudioDeviceID *device);
 #endif
 
@@ -71,7 +71,7 @@
 int64_t ca_frames_to_ns(struct ao *ao, uint32_t frames);
 int64_t ca_get_latency(const AudioTimeStamp *ts);
 
-#if HAVE_COREAUDIO || HAVE_AVFOUNDATION
+#if HAVE_COREAUDIO
 bool ca_stream_supports_compressed(struct ao *ao, AudioStreamID stream);
 OSStatus ca_lock_device(AudioDeviceID device, pid_t *pid);
 OSStatus ca_unlock_device(AudioDeviceID device, pid_t *pid);

diff --git forkSrcPrefix/audio/out/ao_coreaudio_utils.c forkDstPrefix/audio/out/ao_coreaudio_utils.c
--- forkSrcPrefix/audio/out/ao_coreaudio_utils.c
+++ forkDstPrefix/audio/out/ao_coreaudio_utils.c
@@ -28,14 +28,14 @@
 #include "audio/format.h"
 #include "osdep/mac/compat.h"
 
-#if HAVE_COREAUDIO || HAVE_AVFOUNDATION
+#if HAVE_COREAUDIO
 #include "audio/out/ao_coreaudio_properties.h"
 #include <CoreAudio/HostTime.h>
 #else
 #include <mach/mach_time.h>
 #endif
 
-#if HAVE_COREAUDIO || HAVE_AVFOUNDATION
+#if HAVE_COREAUDIO
 static bool ca_is_output_device(struct ao *ao, AudioDeviceID dev)
 {
     size_t n_buffers;
@@ -300,7 +300,7 @@
 
 int64_t ca_get_latency(const AudioTimeStamp *ts)
 {
-#if HAVE_COREAUDIO || HAVE_AVFOUNDATION
+#if HAVE_COREAUDIO
     uint64_t out = AudioConvertHostTimeToNanos(ts->mHostTime);
     uint64_t now = AudioConvertHostTimeToNanos(AudioGetCurrentHostTime());
 
@@ -323,7 +323,7 @@
 #endif
 }
 
-#if HAVE_COREAUDIO || HAVE_AVFOUNDATION
+#if HAVE_COREAUDIO
 bool ca_stream_supports_compressed(struct ao *ao, AudioStreamID stream)
 {
     AudioStreamRangedDescription *formats = NULL;

diff --git forkSrcPrefix/audio/out/ao_coreaudio_chmap.h forkDstPrefix/audio/out/ao_coreaudio_chmap.h
--- forkSrcPrefix/audio/out/ao_coreaudio_chmap.h
+++ forkDstPrefix/audio/out/ao_coreaudio_chmap.h
@@ -30,6 +30,8 @@
 AudioChannelLayout *ca_find_standard_layout(void *talloc_ctx, AudioChannelLayout *l);
 AudioChannelLayout *ca_get_acl(struct ao *ao, size_t *out_layout_size);
 void ca_log_layout(struct ao *ao, int l, AudioChannelLayout *layout);
+#endif
+#if HAVE_COREAUDIO
 bool ca_init_chmap(struct ao *ao, AudioDeviceID device);
 void ca_get_active_chmap(struct ao *ao, AudioDeviceID device, int channel_count,
                          struct mp_chmap *out_map);

diff --git forkSrcPrefix/audio/out/ao_coreaudio_chmap.c forkDstPrefix/audio/out/ao_coreaudio_chmap.c
--- forkSrcPrefix/audio/out/ao_coreaudio_chmap.c
+++ forkDstPrefix/audio/out/ao_coreaudio_chmap.c
@@ -22,7 +22,6 @@
 #include "ao_coreaudio_utils.h"
 
 #include "ao_coreaudio_chmap.h"
-#include <CoreAudioTypes/CoreAudioTypes.h>
 
 static const int speaker_map[][2] = {
     { kAudioChannelLabel_Left,                 MP_SPEAKER_ID_FL   },
@@ -337,7 +336,9 @@
         *out_layout_size = layout_size;
     return layout;
 }
+#endif
 
+#if HAVE_COREAUDIO
 
 #define CHMAP(n, ...) &(struct mp_chmap) MP_CONCAT(MP_CHMAP, n) (__VA_ARGS__)
 

diff --git forkSrcPrefix/audio/out/ao_coreaudio_properties.h forkDstPrefix/audio/out/ao_coreaudio_properties.h
--- forkSrcPrefix/audio/out/ao_coreaudio_properties.h
+++ forkDstPrefix/audio/out/ao_coreaudio_properties.h
@@ -19,6 +19,10 @@
 #ifndef MPV_COREAUDIO_PROPERTIES_H
 #define MPV_COREAUDIO_PROPERTIES_H
 
+#include "config.h"
+
+#if HAVE_COREAUDIO
+
 #include <AudioToolbox/AudioToolbox.h>
 
 #include "internal.h"
@@ -58,4 +62,5 @@
 
 #define CA_SETTABLE(id, sel, data) ca_settable(id, CA_GLOBAL, sel, data)
 
+#endif /* HAVE_COREAUDIO */
 #endif /* MPV_COREAUDIO_PROPERTIES_H */

diff --git forkSrcPrefix/audio/out/ao_coreaudio_properties.c forkDstPrefix/audio/out/ao_coreaudio_properties.c
--- forkSrcPrefix/audio/out/ao_coreaudio_properties.c
+++ forkDstPrefix/audio/out/ao_coreaudio_properties.c
@@ -19,6 +19,10 @@
  * Abstractions on the CoreAudio API to make property setting/getting suck less
 */
 
+#include "config.h"
+
+#if HAVE_COREAUDIO
+
 #include "audio/out/ao_coreaudio_properties.h"
 #include "audio/out/ao_coreaudio_utils.h"
 #include "mpv_talloc.h"
@@ -101,3 +105,4 @@
 
     return AudioObjectIsPropertySettable(id, &p_addr, data);
 }
+#endif /* HAVE_COREAUDIO */

diff --git forkSrcPrefix/audio/out/ao_avfoundation.m forkDstPrefix/audio/out/ao_avfoundation.m
--- forkSrcPrefix/audio/out/ao_avfoundation.m
+++ forkDstPrefix/audio/out/ao_avfoundation.m
@@ -249,9 +249,11 @@
         goto error;
     }
 
+#if TARGET_OS_MAC && !TARGET_OS_IPHONE
     if (ao->device && ao->device[0]) {
         [p->renderer setAudioOutputDeviceUniqueID:(NSString*)cfstr_from_cstr(ao->device)];
     }
+#endif
 
     [p->synchronizer addRenderer:p->renderer];
 #if HAVE_MACOS_11_3_FEATURES
@@ -362,6 +364,8 @@
     .reset          = stop,
     .start          = start,
     .set_pause      = set_pause,
+#if TARGET_OS_MAC && !TARGET_OS_IPHONE
     .list_devs      = ca_get_device_list,
+#endif
     .priv_size      = sizeof(struct priv),
 };

diff --git forkSrcPrefix/meson.build forkDstPrefix/meson.build
--- forkSrcPrefix/meson.build
+++ forkDstPrefix/meson.build
@@ -821,6 +821,9 @@
 if features['avfoundation']
     dependencies += avfoundation
     sources += files('audio/out/ao_avfoundation.m')
+    if not features['cocoa']
+        sources += files('osdep/utils-mac.c')
+    endif
 endif
 
 coreaudio = dependency('appleframeworks', modules: ['CoreFoundation', 'CoreAudio',
